emulation.c

```diff
@@ -436,6 +436,23 @@ void __stdcall return_to_host() {
 Address CreateHlt() {
 #ifdef UC_NATIVE
 
+  unsigned int code_size = 20;
+
+#if 0
+  Address code_address = Allocate(20);
+#else
+  static Address buffer;
+  static int buffer_size = 0;
+  if (buffer_size < code_size) {
+    buffer_size = 0x1000;
+    buffer = Allocate(buffer_size);
+  }
+  Address code_address = buffer;
+  buffer += code_size;
+  buffer_size -= code_size;
+#endif
```

Comment:
Should be moved into the allocator itself, so other code can also benefit (for small allocations).

It isn't necessary if the allocator has smaller pages.